app/auth/authV2/actions/SessionTimeoutPredicateV2.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package auth.authV2.actions
+
+import play.api.Logger
+import play.api.mvc.Results.Redirect
+import play.api.mvc._
+import uk.gov.hmrc.http.SessionKeys
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
```

Comment:
I would assume we need a new version for SessionTimeoutPredicate (V2?) in order to break dependency from BaseController and make it properly composable with the rest of the actions we have?

Reply:
Yes, where-ever I needed to modify and existing class, I made a duplicate of it.